enterprise/api/v1/views/enterprise_customer_sso_configuration.py

```diff
@@ -59,3 +91,114 @@ def oauth_orchestration_complete(self, request, configuration_uuid, *args, **kwa
         sso_configuration_record.active = True
         sso_configuration_record.save()
         return Response(status=HTTP_200_OK)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def retrieve(self, request, *args, **kwargs):
+        sso_configuration_record = fetch_configuration_record(kwargs)
+        if not sso_configuration_record:
+            return Response(status=HTTP_404_NOT_FOUND)
+        # Make sure the requesting user is part of the enterprise customer associated with the configuration record
+        # or are a staff member
+        config_customer = sso_configuration_record.first().enterprise_customer
+        if not check_user_part_of_customer(request.user, config_customer):
+            return Response(status=HTTP_404_NOT_FOUND)
+
+        serializer = self.serializer_class(sso_configuration_record.first())
+        return Response(serializer.data)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def list(self, request, *args, **kwargs):
+        list_queryset = self.queryset
+
+        # If the request includes a customer uuid, we require through permissioning that the requesting user is an
+        # admin of that customer
+        if provided_customer := request.query_params.get('enterprise_customer'):
+            enterprise_customer = EnterpriseCustomer.objects.filter(pk=provided_customer)
+            list_queryset = list_queryset.filter(enterprise_customer__in=enterprise_customer)
+
+        # For non-staff users, only show the configurations that are connected to enterprise customers
+        # that they are associated with
+        if not request.user.is_staff:
+            ent_customers = EnterpriseCustomer.objects.filter(
+                enterprise_customer_users__in=EnterpriseCustomerUser.objects.filter(user_id=request.user.id)
+            )
+            list_queryset = list_queryset.filter(enterprise_customer__in=ent_customers)
+
+        serializer = self.serializer_class(list_queryset, many=True)
+        return Response(serializer.data, status=HTTP_200_OK)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def create(self, request, *args, **kwargs):
+        # Force the enterprise customer to be the one associated with the user
+        request_data = request.data.dict().copy()
+        requesting_user_customer = request_data.get('enterprise_customer')
+        if requesting_user_customer:
+            try:
+                enterprise_customer = EnterpriseCustomer.objects.get(uuid=requesting_user_customer)
+            except EnterpriseCustomer.DoesNotExist:
+                return Response({'error': BAD_CUSTOMER_ERROR}, status=HTTP_400_BAD_REQUEST)
```

Comment:
i'd probably go for a 403 or 404 here rather than a plain 400 - the request is valid, we just cannot find the resource (in this case we may not want them to be able to disambiguate between not found and not allowed)

Reply:
uhhh 404 is probably best here?